toolkit/tools/grapher/grapher.go

```diff
@@ -217,15 +250,16 @@ func addSingleDependency(g *pkggraph.PkgGraph, packageNode *pkggraph.PkgNode, de
 		return err
 	}
 
-	if nodes == nil {
-		dependentNode, err = addUnresolvedPackage(g, dependency)
+	// If we can't find the dependency in the graph, or it is a remote dependency, we need to do a bit of extra validation.
+	if nodes == nil || nodes.RunNode.Type != pkggraph.TypeLocalRun {
+		dependentNode, err = handleRemoteDependency(g, dependency)
 		if err != nil {
-			err = fmt.Errorf("failed to add a package (%s):\n%w", dependency.Name, err)
-			return err
+			err = fmt.Errorf("failed to handle remote dependency from %+v to %+v:\n%w", packageNode.VersionedPkg, dependency, err)
```

Comment:
Shouldn't we return the `err` here?
```suggestion
			err = fmt.Errorf("failed to handle remote dependency from %+v to %+v:\n%w", packageNode.VersionedPkg, dependency, err)
			return err
```